pyspark_ai/spark_sql_chain.py

```diff
@@ -0,0 +1,70 @@
+from typing import Any, Optional, List
+
+from langchain.callbacks.base import Callbacks
+from langchain.chains import LLMChain
+from langchain.chat_models.base import BaseChatModel
+from langchain.schema import HumanMessage, BaseMessage
+from pyspark.sql import SparkSession
+from pyspark_ai.code_logger import CodeLogger
+
+from pyspark_ai.ai_utils import AIUtils
+
+
+class SparkSQLChain(LLMChain):
```

Comment:
TODO: we can abstract the code of SparkSQLChain and PythonExecutor to a new common super class.

Reply:
Caching is the major difference here. I will unify them in a follow-up PR.